active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -0,0 +1,247 @@
+
+This is especially problematic for 3rd party namespace identifiers specified in a `bicepconfig.json` up a file hierarchy, since they would result in a compilation error because they are unspecified in the builtin configuration.
+
+1. TODO: Potential drawback when trying to inspect module sources 
+<!--
+Good morning Stephen & Anthony, I am currently working on a design draft for a feature proposal that may have an overlap with the work you are doing with publishing of sources. The proposal highlights are described here: Proposal - Simplification of "built-in" vs "registry-supplied" providers · Issue #12202 · Azure/bicep (github.com)
+ 
+One potential consideration is how would the identifiers be resolved when a user is looking at module sources, specifically consider the scenario below. 
+Stephen, I would really appreciate your thoughts on whether this new syntax affects the source publish requirements  or has some other kind of implications in the ability to resolve published sources in the editor.
+ 
+Scenario: User creates and publishes with sources using the new provider syntax 
+ 
+Question: How does the VSCode extension  resolve the registry location and provider version it needs to resolve the namespace identifier (`az`) when the consumer of this module want's to inspect the sources?
+ 
+my_mod.bicep
+provider az
+param location string = 'eastus'
+ 
+resource sa 'Microsoft.Storage/storageAccounts@2023-01-01' = {
+  name: 'test'
+  location: location
+  kind: 'StorageV2'
+  sku: {
+    name: 'Standard_LRS'
+  }
+}
+
+
+Let's say that the project where the user is publishing this module had the following bicepconfig.json:
+bicepconfig.json
+{
+  "providers": {
+    "az": {
+      "source": {
+        "registry": "mcr.microsoft.com/bicep/providers/az",
+        "version": "0.2.3",
+      },      "isDefaultImport": true    },    
+      "sys": {
+          "source": {
+          "builtin": true
+      },
+      "isDefaultImport": true
+    }
+  }
+}
+-->
+
+## Alternatives
+
+- Using a separate file other than `bicepconfig.json` to store resource type provider version information was evaluated but adds complexity to UX as users need to be aware of two separate merge semantics (one for config and one for provider packages) 
+
+## Rollout plan
+
+This change affects the client side only and is applied only when the pre-existing feature flag `DynamicTypeLoadingEnabled` is set. 
+
+## Unresolved questions
+
+> - What aspects of the design do you expect to resolve through the REP process before merging?
+> - What parts of the design do you anticipate resolving through feature implementation before stabilization?
+
```

Comment:
One open question for me here is, what about version matchers and lock files? That seems like a question for a future REP but one we should be mindful of lest any one-way doors be taken in implementing this REP.